File: DSA/Arrays/carry-forward/best-time-buy-sell-stocks/javascript/bestTimeToSellStocks.js
const bestTimeToSellStocks = (A) => {
    let min = A[0];
    let profit = 0;
    A.map((ele) => {
        if (ele < min) {
            min = ele;
        } else {
            if (profit < ele - min) {
                profit = ele - min;
            }
        }
    });
    return profit;
};

File: DSA/Arrays/carry-forward/best-time-buy-sell-stocks/python/bestTimeToSellStocks.py
def maxProfit(self, A):
        if len(A) < 1:
            return 0
        min = A[0] 
        profit = 0
        for i in A:
            if min > i:
                min = i
            else: 
                profit = max(profit, i - min)
        return profit